### DISCUSSION TITLE
Template rendering issue for custom bundle

### DISCUSSION THREAD

**Discussion by ghost** [pimcore/pimcore#15628] [Q&A / Support]
I am working on custom bundle named CompareObjectBundle and trying to render a view file named compareObjects.html.twig using below code:
![Screenshot from 2023-07-26 15-17-52](https://github.com/pimcore/pimcore/assets/116332084/d301aed9-204b-4ed4-bf77-005f6292b541)

return $this->render('CompareObjectBundle:Default:compareObjects.html.twig', $result);
I have tried like this too :
return $this->render('CompareObjectBundle/Default/compareObjects.html.twig', $result);
but getting below error :
Unable to find template "CompareObjectBundle:Default:compareObjects.html.twig" (looked into: /var/www/html/project/templates, /var/www/html/project/vendor/symfony/twig-bridge/Resources/views/Form).

The actual path of my view file is:
/src/Acme/CompareObjectBundle/Resources/views/compareObjects.html.twig

I have attached the screenshots related to this , kindly have a look into this and help me in knowing what is wrong with the code.

**Comment by kingjia90** [comment]
Please try `$this->render('@CompareObjectBundle/compareObjects.html.twig', $result);` 

See https://symfony.com/doc/current/templates.html#bundle-templates

**Comment by blankse** [comment]
I think it should be:
`return $this->render('@CompareObject/compareObjects.html.twig', $result);`

**Reply by kingjia90** [reply]
Oh, that’s right, the documentation says it’s without ‘Bundle’ suffix

**Comment by ghost** [comment]
Thanks a lot for the provided solutions but these are not working for me.

As per article reference given :

 **If this bundle includes the template <your-project>/vendor/acmefoo-bundle/Resources/views/user/profile.html.twig, you can refer to it as @AcmeFoo/user/profile.html.twig.**
 
 As I mentioned earlier my bundle is a custom bundle and not a bundle to be reused. Hence , it does not have vendor folder. Kindly let me know how twig template will be picked from the bundle directory itself. Kindly check my original question post for the path and how I am rendering it.
 
 Thank you very much in advance.

**Reply by blankse** [reply]
@aniketdibbe Is your CompareObjectBundle enabled? Let us show the result of the ListCommand:
`php bin/console pimcore:bundle:list`

and the content of `config/bundles.php`

**Reply by ghost** [reply]
yes it's already enabled.

**Reply by blankse** [reply]
In the ListCommand result the bundle name ist also "CompareObjectBundle"?

**Reply by ghost** [reply]
yes its AcmeCompareObjectBundle

**Reply by blankse** [reply]
Than you need this:
`return $this->render('@AcmeCompareObject/compareObjects.html.twig', $result);`

**Reply by ghost** [reply]
Thanks now its's working.

**Comment by ghost** [comment]
I wanted to extend my sincerest thanks for your help in solving my query.Your expertise and willingness to assist me are truely appreciated.Thanks everyone.
